platform/broadcom/docker-syncd-brcm/Dockerfile.j2

```diff
@@ -15,8 +15,24 @@ debs/{{ deb }}{{' '}}
 {%- endfor -%}
 debs/
 
+# Install dependencies for libsaibcm debian packages
+{%- set syncd_brcm_debs = [] %}
+{% for deb in docker_syncd_brcm_debs.split(' ') -%}
+{%- if 'libsaibcm' in deb %}
+RUN apt-get install -y $(dpkg --info debs/{{ deb }} | grep Depends | sed "s/.*ends:\ //" | tr '\n' ' ' | tr ',' ' ')
+{%- else %}
+{{- syncd_brcm_debs.append(deb) or '' }}
+{%- endif %}
+{%- endfor %}
+
+# To store saibcm debian packages
+RUN mkdir /usr/local/debs
+
+# Move the saibcm debian packages
+RUN mv debs/libsaibcm*_amd64.deb /usr/local/debs
+
 # Install locally-built Debian packages and implicitly install their dependencies
-{{ install_debian_packages(docker_syncd_brcm_debs.split(' ')) }}
+{{ install_debian_packages(syncd_brcm_debs) }}
```

Comment:
i thought the plan was to install both dnx and xgs debs, and use symbol link to choose which library to load during the syncd run time.

Reply:
Since both the xgs/dnx debian packages keeps the SAI shared object file in same location it would get overwritten if we install both. Also the docker image could grow big if we install both debs.

jujoseph@jujoseph-Virtual-Machine:~/3164/SAI/v.4.3$ sudo dpkg -i libsaibcm_4.3.3.1_amd64.deb
jujoseph@jujoseph-Virtual-Machine:~/3164/SAI/v.4.3$ ls -alh /usr/lib/libsai.so.1.0
-rw-r--r-- 1 root root **225M** Feb 26 10:32 /usr/lib/libsai.so.1.0

jujoseph@jujoseph-Virtual-Machine:~/3164/SAI/v.4.3$ sudo dpkg -i libsaibcm-dnx_4.3.3.1_amd64.deb 
jujoseph@jujoseph-Virtual-Machine:~/3164/SAI/v.4.3$ ls -alh /usr/lib/libsai.so.1.0
-rw-r--r-- 1 root root **353M** Feb 26 10:32 /usr/lib/libsai.so.1.0

So the idea was to copy the xgs and dnx debs into docker image. Install the correct deb package based on ASIC family the first time it creates the syncd docker container and delete the debs in the container. The dependent debian packages are installed during syncd docker build time. Let me know if approach is ok.

Reply:
not sure how long this could take for the first reboot. if we can rename the dnxsai to libsaidnx.so.1.0, and xgssai to libsaixgs.so.1.0, then we can use symbol link to decide which one to choose, it can reduce the first time reboot time.

Reply:
I had verified this, on an average "dpkg -i libsaibcm_4.3.3.1_amd64.deb" takes around 5 secs on this Arista box.
1. real    0m5.358s
2. real    0m5.096s
3. real    0m5.114s

We could install both the libsai packages as well, as you suggested - but I was more worried about the increase in size of the syncd/sonic-buildimage as **Each libsai.so.1.0 is around 350MB** ( hopefully since we do gzip and docker save, we are ok - **I will confirm**). 

The size was the factor why I was looking for keeping the libsai and libsai-dnx debs as such .. and installing them later in syncd container during runtime.

Reply:
I did some more experiments here, I find if we install both the xgs/dnx debs and rename the libsai.so.1.0 - the size of the syncd docker image on DUT is going high to 1GB 

Case 1: Copy the debs, don't install during build- install the correct SAI debian on boot up 
```
-rw-r--r-- 1 jujoseph jujoseph 243M Mar 11 08:59 target/docker-syncd-brcm.gz
-rwxr-xr-x 1 jujoseph jujoseph 906M Mar 11 09:46 target/sonic-aboot-broadcom.swi
```
On the DUT
```
docker-syncd-brcm             latest              b55d61bd0b6a        4 hours ago         514MB
```

Case 2: Install both debs, during build but rename the libsai.so with xgs and dnx
```
-rw-r--r-- 1 jujoseph jujoseph 285M Mar 10 22:38 docker-syncd-brcm.gz
-rwxr-xr-x 1 jujoseph jujoseph 951M Mar 10 23:29 target/sonic-aboot-broadcom.swi
```
On the DUT
```
docker-syncd-brcm             dnx.0-dirty-20210310.224235   dee3556e3447        24 hours ago        1.05GB
```

Reply:
More findings on investigation of whether installing the module adds more time in getting interfaces UP there by affecting control plane traffic in fast reload.. The following tests are done with first cold boot but I believe similar time consumed in first fast boot as well.

The test is basically to check how long from syncd docker create/start does it take for interfaces to be UP on bootup.

**Default behaviour**
syands starts - lets us say time T
syncd ( interfaces up ) ->   (T+53 secs) / (T+51 secs) / (T+49 secs) 

**With the libsaibcm installation during syncd docker start** 
syands starts - lets us say time T
syncd ( interfaces up ) ->   (T+51 secs) / (T+50 secs) / (T+53 secs) 

The conclusion is that even though we take additional 4-5sec for installing the libsaibcm , in both cases, we see the interfaces comes up in range of T+49 sec - T+53 sec, where T is the time when syncd container is created/started.

An additional optimization I plan is to 
1. Keep the libsaibcm for XGS installed defaultly in the syncd docker - **so that it won't affect the existing XGS platforms** 
2. Only if the platform is DNX based, remove and install the new libsaibcm-DNX debian package.